### ISSUE TITLE
Short contig issue of TIR-Learner.sh

### ISSUE THREAD

**Issue by aderzelle** [oushujun/EDTA#14]
Hello,
I am rerunning the last push in the same folder and get errors, here is the STDOUT and STDERR
This is a follow-up of this issue: 
https://github.com/oushujun/EDTA/issues/10

```
./EDTA/EDTA.pl -genome Avaga.Masurca.Graal.min5500.fa -species others -step all -t 48 2>&1 |tee EDTA.log

########################################################
##### Extensive de-novo TE Annotator (EDTA) v1.5    ####
##### Shujun Ou (shujun.ou.1@gmail.com)             ####
########################################################



Mo Aug 19 10:52:23 CEST 2019	Dependency checking:
				All passed!
Mo Aug 19 10:52:33 CEST 2019	Obtain raw TE libraries using various structure-based programs: 
Mo Aug 19 10:52:33 CEST 2019	EDTA_raw: Check files and dependencies, prepare working directories.

Mo Aug 19 10:52:33 CEST 2019	Start to find LTR candidates.

Mo Aug 19 10:52:33 CEST 2019	Existing result file Avaga.Masurca.Graal.min5500.fa.LTRlib.fa found! Will keep this file without rerunning this module.
	Please specify -overwrite 1 if you want to rerun this module.

Mo Aug 19 10:52:33 CEST 2019	Finish finding LTR candidates.

Mo Aug 19 10:52:33 CEST 2019	Start to find TIR candidates.

Mo Aug 19 10:52:33 CEST 2019	Identify TIR candidates from scratch.

Species: others
/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.22/TIR-Learner.sh: 95: [: others: unexpected operator
/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.22/TIR-Learner.sh: 95: [: others: unexpected operator
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.22/Module3_New/getDataset2.py", line 107, in Predict
    model = load_model(path+"/Module3_New/"+'CNN0724.h5')
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/engine/saving.py", line 249, in load_model
    optimizer_config, custom_objects=custom_objects)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 838, in deserialize
    printable_module_name='optimizer')
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 194, in deserialize_keras_object
    return cls.from_config(cls_config)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 159, in from_config
    return cls(**config)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 471, in __init__
    super(Adam, self).__init__(**kwargs)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 68, in __init__
    'passed to optimizer: ' + str(k))
TypeError: Unexpected keyword argument passed to optimizer: name
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.22/Module3_New/getDataset2.py", line 131, in <module>
    d = pool.map(Predict,files)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
TypeError: Unexpected keyword argument passed to optimizer: name
cat: '*-+-DTA.fa': No such file or directory
cat: '*-+-DTC.fa': No such file or directory
cat: '*-+-DTH.fa': No such file or directory
cat: '*-+-DTM.fa': No such file or directory
cat: '*-+-DTT.fa': No such file or directory
cat: '*-+-NonTIR.fa': No such file or directory
cat: '*-+-*-+-*.gff3': No such file or directory
rm: cannot remove '*-+-*-+-*.gff3': No such file or directory
Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.22/Module3_New/CombineAll.py", line 90, in <module>
    keep=removeIRFhomo("%s.gff3"%(genome_Name+spliter+dataset),remove,"%sClean.gff3"%(genome_Name+spliter+dataset+spliter))
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.22/Module3_New/CombineAll.py", line 76, in removeIRFhomo
    f=pd.read_csv(file,header=None,sep="\t")
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 702, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 429, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 895, in __init__
    self._make_engine(self.engine)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 1122, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 1853, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 545, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file
Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.22/Module3/GetAllSeq.py", line 62, in <module>
    file=open(f,"r+")
FileNotFoundError: [Errno 2] No such file or directory: 'TIR-Learner_FinalAnn.gff3'
mv: cannot stat 'TIR-Learner/*FinalAnn.gff3': No such file or directory
mv: cannot stat 'TIR-Learner/*FinalAnn.fa': No such file or directory
Can't open ./TIR-Learner-Result/TIR-Learner_FinalAnn.fa: No such file or directory at /media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/util/rename_tirlearner.pl line 18.
Warning: LOC list Avaga.Masurca.Graal.min5500.fa.TIR.ext30.list is empty.
Warning: The TIR result file has 0 bp!

Mo Aug 19 10:52:56 CEST 2019	Start to find MITE candidates.

Mo Aug 19 10:52:56 CEST 2019	Existing result file Avaga.Masurca.Graal.min5500.fa.MITE.raw.fa found! Will keep this file without rerunning this module.
	Please specify -overwrite 1 if you want to rerun this module.

Mo Aug 19 10:52:56 CEST 2019	Finish finding MITE candidates.

Mo Aug 19 10:52:56 CEST 2019	Start to find Helitron candidates.

Mo Aug 19 10:52:56 CEST 2019	Existing result file Avaga.Masurca.Graal.min5500.fa.Helitron.raw.fa found! Will keep this file without rerunning this module.
	Please specify -overwrite 1 if you want to rerun this module.

Mo Aug 19 10:52:56 CEST 2019	Finish finding Helitron candidates.
```

**Comment by oushujun** [comment]
Hi, this seems to system specific. Can you provide your system
configuration info? Thanks!

**Comment by aderzelle** [comment]
yes, it's Ubuntu 18.04. What kind of information would you need?

**Comment by oushujun** [comment]
I updated the code and hopefully resolve this issue. Please update EDTA and run again. No need to reinstall. Just go to the EDTA folder and `git pull`. Thanks!

**Comment by aderzelle** [comment]
Ok I will do it,
there are many things running on my cluster, it might take a little while but I will update you.

**Comment by aderzelle** [comment]
Hello, I am getting an other error

```
./EDTA/EDTA.pl -genome Avaga.Masurca.Graal.min5500.fa -species others -step all -t 28 2>&1 |tee EDTA.log
cat EDTA.log 

########################################################
##### Extensive de-novo TE Annotator (EDTA) v1.5    ####
##### Shujun Ou (shujun.ou.1@gmail.com)             ####
########################################################



Di Aug 27 21:05:30 CEST 2019	Dependency checking:
				All passed!
Di Aug 27 21:05:44 CEST 2019	Obtain raw TE libraries using various structure-based programs: 
Di Aug 27 21:05:44 CEST 2019	EDTA_raw: Check files and dependencies, prepare working directories.

Di Aug 27 21:05:44 CEST 2019	Start to find LTR candidates.

Di Aug 27 21:05:44 CEST 2019	Identify LTR retrotransposon candidates from scratch.

Di Aug 27 21:14:39 CEST 2019	Finish finding LTR candidates.

Di Aug 27 21:14:39 CEST 2019	Start to find TIR candidates.

Di Aug 27 21:14:39 CEST 2019	Identify TIR candidates from scratch.

Species: others
rm: cannot remove './TIR-Learner/*': No such file or directory
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3_New/getDataset2.py", line 106, in Predict
    model = load_model(path+"/Module3_New/"+'CNN0724.h5')
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/engine/saving.py", line 249, in load_model
    optimizer_config, custom_objects=custom_objects)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 838, in deserialize
    printable_module_name='optimizer')
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 194, in deserialize_keras_object
    return cls.from_config(cls_config)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 159, in from_config
    return cls(**config)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 471, in __init__
    super(Adam, self).__init__(**kwargs)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 68, in __init__
    'passed to optimizer: ' + str(k))
TypeError: Unexpected keyword argument passed to optimizer: name
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3_New/getDataset2.py", line 130, in <module>
    d = pool.map(Predict,files)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
TypeError: Unexpected keyword argument passed to optimizer: name
cat: '*-+-DTA.fa': No such file or directory
cat: '*-+-DTC.fa': No such file or directory
cat: '*-+-DTH.fa': No such file or directory
cat: '*-+-DTM.fa': No such file or directory
cat: '*-+-DTT.fa': No such file or directory
cat: '*-+-NonTIR.fa': No such file or directory
cat: '*-+-*-+-*.gff3': No such file or directory
rm: cannot remove '*-+-*-+-*.gff3': No such file or directory
Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3_New/CombineAll.py", line 90, in <module>
    keep=removeIRFhomo("%s.gff3"%(genome_Name+spliter+dataset),remove,"%sClean.gff3"%(genome_Name+spliter+dataset+spliter))
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3_New/CombineAll.py", line 76, in removeIRFhomo
    f=pd.read_csv(file,header=None,sep="\t")
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 702, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 429, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 895, in __init__
    self._make_engine(self.engine)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 1122, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 1853, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 545, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3/GetAllSeq.py", line 32, in GetListFromFile
    f=open(file,"r+")
FileNotFoundError: [Errno 2] No such file or directory: 'TIR-Learner_FinalAnn.gff3'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3/GetAllSeq.py", line 63, in <module>
    pool.map(GetListFromFile,fileList) #shujun
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
FileNotFoundError: [Errno 2] No such file or directory: 'TIR-Learner_FinalAnn.gff3'
mv: cannot stat 'TIR-Learner/*FinalAnn.gff3': No such file or directory
mv: cannot stat 'TIR-Learner/*FinalAnn.fa': No such file or directory
Can't open ./TIR-Learner-Result/TIR-Learner_FinalAnn.fa: No such file or directory at /media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/util/rename_tirlearner.pl line 18.
Warning: LOC list Avaga.Masurca.Graal.min5500.fa.TIR.ext30.list is empty.
Warning: The TIR result file has 0 bp!

Di Aug 27 21:54:58 CEST 2019	Start to find MITE candidates.

Di Aug 27 21:54:58 CEST 2019	Identify MITE candidates from scratch.

Di Aug 27 21:54:58 CEST 2019	Warning: Because MITE-Hunter is too slow and only contribute limited new TIR candidates, it is taken down temporary until a better solution is found.

As a temporary fix, the TIR-Learner is used to mock the MITE-Hunter result. Please run -type tir first.

Error: MITE results not found!

ERROR: Raw TIR results not found in Avaga.Masurca.Graal.min5500.fa.EDTA.raw/Avaga.Masurca.Graal.min5500.fa.TIR.raw.fa at ./EDTA/EDTA.pl line 177.
```

**Comment by aderzelle** [comment]
may I kindly bump up this thread?

**Comment by oushujun** [comment]
Sorry for the delay response. We identified the issue which is likely due to the short contigs in genomes. It appears that the previous issue is not fully resolved and still persists when used in fragmented genomes. We are working to resolve the bug and hope to update it in the next version.

Best,
Shujun

**Comment by aderzelle** [comment]
Hello,
I removed the short contings in my assembly, here is the size of the 6 contigs I have, in bp
```
Chrom_3	26566722
Chrom_1	20113852
Chrom_5	19909909
Chrom_2	17221235
Chrom_4	17025431
Chrom_6	15550512
```
so around 20 Mb each. I don't know how "short" is "short" but thus doesn't solve the bug.

**Comment by aderzelle** [comment]
However, I am able to run EDTA on the second computer ... So I guess the bug is hardware dependant?

**Comment by oushujun** [comment]
Hi @aderzelle, thanks for reporting the results. Can you describe the two platforms you are using? Such as hardware, OS, conda version, and etc. For the first computer that fails, have you tried creating a new environment and fresh install the EDTA again?

It's totally fine that the .stat file is empty, which means the library is non-redundant already. Depending on the kind of stats you want, you may want to do different things. The `genome.FLYE.sixLongest.fa.EDTA.TElib.fa` only contains non-redundant TE examplars. If you want to estimate the copy number or the size of different TEs in the library, you may want to use RepeatMasker with the custom EDTA library to do so.

Best,
Shujun

**Comment by aderzelle** [comment]
Failure machine:
Ubuntu 18.04.3 LTS
conda 4.6.14
Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz

Working machine:
Ubuntu 18.10
conda 4.7.11
Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz

**Comment by oushujun** [comment]
I have no clue, sorry. Hardware seems the same. Maybe you can update conda
on the failure machine and reinstall EDTA.

**Comment by aderzelle** [comment]
Yes, I haven't fixed it yet but it's most probably a problem with the python install.
So I am closing the issue for now. 
Thank you for the support.

**Comment by aderzelle** [comment]
Hello,
just as an update, the python install is all right, I am still having the following issue. I have tried reinstalling from scratch 

```
rm: cannot remove './TIR-Learner/*': No such file or directory
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3_New/getDataset2.py", line 106, in Predict
    model = load_model(path+"/Module3_New/"+'CNN0724.h5')
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/engine/saving.py", line 249, in load_model
    optimizer_config, custom_objects=custom_objects)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 838, in deserialize
    printable_module_name='optimizer')
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 194, in deserialize_keras_object
    return cls.from_config(cls_config)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 159, in from_config
    return cls(**config)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 471, in __init__
    super(Adam, self).__init__(**kwargs)
  File "/home/urbe/.local/lib/python3.6/site-packages/tensorflow/python/keras/optimizers.py", line 68, in __init__
    'passed to optimizer: ' + str(k))
TypeError: Unexpected keyword argument passed to optimizer: name
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3_New/getDataset2.py", line 130, in <module>
    d = pool.map(Predict,files)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
TypeError: Unexpected keyword argument passed to optimizer: name
cat: '*-+-DTA.fa': No such file or directory
cat: '*-+-DTC.fa': No such file or directory
cat: '*-+-DTH.fa': No such file or directory
cat: '*-+-DTM.fa': No such file or directory
cat: '*-+-DTT.fa': No such file or directory
cat: '*-+-NonTIR.fa': No such file or directory
cat: '*-+-*-+-*.gff3': No such file or directory
rm: cannot remove '*-+-*-+-*.gff3': No such file or directory
Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3_New/CombineAll.py", line 90, in <module>
    keep=removeIRFhomo("%s.gff3"%(genome_Name+spliter+dataset),remove,"%sClean.gff3"%(genome_Name+spliter+dataset+spliter))
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3_New/CombineAll.py", line 76, in removeIRFhomo
    f=pd.read_csv(file,header=None,sep="\t")
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 685, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 457, in _read
    parser = TextFileReader(fp_or_buf, **kwds)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 895, in __init__
    self._make_engine(self.engine)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 1135, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/site-packages/pandas/io/parsers.py", line 1917, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 545, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3/GetAllSeq.py", line 32, in GetListFromFile
    f=open(file,"r+")
FileNotFoundError: [Errno 2] No such file or directory: 'TIR-Learner_FinalAnn.gff3'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/bin/TIR-Learner1.23/Module3/GetAllSeq.py", line 63, in <module>
    pool.map(GetListFromFile,fileList) #shujun
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/urbe/anaconda3/envs/EDTA/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
FileNotFoundError: [Errno 2] No such file or directory: 'TIR-Learner_FinalAnn.gff3'
mv: cannot stat 'TIR-Learner/*FinalAnn.gff3': No such file or directory
mv: cannot stat 'TIR-Learner/*FinalAnn.fa': No such file or directory
Can't open ./TIR-Learner-Result/TIR-Learner_FinalAnn.fa: No such file or directory at /media/urbe/MyBDrive/Alessandro/TR_annotation/EDTA/util/rename_tirlearner.pl line 18.
Warning: LOC list A_vaga.FLYE.sixLongest.fa.TIR.ext30.list is empty.
Warning: The TIR result file has 0 bp!

Do Okt  3 15:24:37 CEST 2019	Start to find MITE candidates.

Do Okt  3 15:24:37 CEST 2019	Identify MITE candidates from scratch.

Do Okt  3 15:24:37 CEST 2019	Warning: Because MITE-Hunter is too slow and only contribute limited new TIR candidates, it is taken down temporary until a better solution is found.

As a temporary fix, the TIR-Learner is used to mock the MITE-Hunter result. Please run -type tir first.

Error: MITE results not found!

ERROR: Raw TIR results not found in A_vaga.FLYE.sixLongest.fa.EDTA.raw/A_vaga.FLYE.sixLongest.fa.TIR.raw.fa at ./EDTA/EDTA.pl line 177.

```
The only modification from the EDTA install is that I did not install repeatmodeler and repeatmasker through bioconda ( but they are in my $PATH and working).

**Comment by oushujun** [comment]
This bug should be resolved in EDTA v1.6.

**Comment by oushujun** [comment]
@AntoineHo Thank you for the tip! I have included it in the debugging Q&A: https://github.com/oushujun/EDTA/wiki/Installations,-builds,-and-tests-Q&A
